core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1233,16 +1223,13 @@ class DAGScheduler(
               logInfo("waiting: " + waitingStages)
               logInfo("failed: " + failedStages)
 
-              // We supply true to increment the epoch number here in case this is a
+              // Increment the epoch number here in case this is a
               // recomputation of the map outputs. In that case, some nodes may have cached
               // locations with holes (from when we detected the error) and will need the
               // epoch incremented to refetch them.
               // TODO: Only increment the epoch number if this is not the first time
```

Comment:
It might be possible to resolve this `TODO` but I haven't thought about it too much yet.